client/signup/steps/design-type-with-store/index.jsx

```diff
@@ -0,0 +1,120 @@
+			{ type: 'grid', label: this.translate( 'A grid of my latest posts' ), image: <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 310 230"><rect x="15" y="15" fill="#E8F0F5" width="280" height="40"/><rect x="114" y="70" fill="#C3EF96" width="82" height="65"/><rect x="15" y="70" fill="#C3EF96" width="82" height="65"/><rect x="213" y="70" fill="#C3EF96" width="82" height="65"/><rect x="114" y="150" fill="#C3EF96" width="82" height="65"/><rect x="15" y="150" fill="#C3EF96" width="82" height="65"/><rect x="213" y="150" fill="#C3EF96" width="82" height="65"/></svg> },
+			{ type: 'store', label: this.translate( 'An online store' ), image: <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 307 229"><g fill="none" fill-rule="evenodd"><path fill="#C4EF96" d="M14 121h82v65H14zM14 201h82v28H14zM210 121h82v65h-82zM210 201h82v28h-82zM112 121h82v65h-82zM112 201h82v28h-82z"/><path fill="#E8F0F5" d="M0 0h307v105H0z"/><path d="M112 13h80v80h-80z"/><path d="M142 79.667c0 3.666-3 6.666-6.667 6.666-3.666 0-6.633-3-6.633-6.666C128.7 76 131.667 73 135.333 73 139 73 142 76 142 79.667zM168.667 73c-3.667 0-6.634 3-6.634 6.667 0 3.666 2.967 6.666 6.634 6.666 3.666 0 6.666-3 6.666-6.666 0-3.667-3-6.667-6.666-6.667zm1.32-16.667c3.123 0 5.83-2.17 6.506-5.22L182 29.667h-46.667v-3.334a6.665 6.665 0 0 0-6.666-6.666H122v6.666h6.667V63a6.665 6.665 0 0 0 6.666 6.667h40A6.665 6.665 0 0 0 168.667 63h-33.334v-6.667h34.654z" fill="#D2DEE6"/></g></svg> },
+		];
+	},
+
+	onPressableStoreStepRef( pressableStoreStep ) {
+		this._pressableStore = pressableStoreStep;
+	},
+
+	renderChoice( choice ) {
+		return (
+			<Card className="design-type-with-store__choice" key={ choice.type }>
+				<a className="design-type-with-store__choice__link" href="#" onClick={ ( event ) => this.handleChoiceClick( event, choice.type ) }>
+					{ choice.image }
+					<h2>{ choice.label }</h2>
+				</a>
+			</Card>
+		);
+	},
+
+	renderChoices() {
+		return (
+			<div className="design-type-with-store__list">
+				{ this.getChoices().map( this.renderChoice ) }
+			</div>
+		);
+	},
+
+	render() {
+		let pressableWrapperClassName = classNames( {
+			'design-type-with-store__pressable-wrapper': true,
+			'is-hidden': ! this.state.showStore,
+		} );
+
+		let sectionWrapperClassName = classNames( {
+			'design-type-with-store__section-wrapper': true,
+			'is-hidden': this.state.showStore,
+		} );
+
+		return (
+			<div className="design-type-with-store">
+				<div className={ pressableWrapperClassName } >
+					<PressableStoreStep { ... this.props } ref={ this.onPressableStoreStepRef }/>
```

Comment:
Wouldn't it be better to have this rendered only when it needs to be shown and not have it in the DOM at all times?

There is already similar logic in the Domains step in Signup and the pattern could possibly be reused here. Code can be found [here](https://github.com/Automattic/wp-calypso/blob/045b256fa4531b0ddc0f376b738e8fdac70a6d92/client/signup/steps/domains/index.jsx#L225).